src/main/java/com/ishland/c2me/common/optimization/chunkserialization/ChunkDataSerializer.java

```diff
@@ -0,0 +1,601 @@
+    private static final byte[] STRING_MISSING_BEDROCK = NbtWriter.getStringBytes("missing_bedrock");
+    private static final byte[] STRING_INDICES = NbtWriter.getStringBytes("Indices");
+    private static final byte[] STRING_SIDES = NbtWriter.getStringBytes("Sides");
+    private static final byte[] STRING_ENTITIES = NbtWriter.getStringBytes("entities");
+    private static final byte[] STRING_LIGHTS = NbtWriter.getStringBytes("Lights");
+    private static final byte[] STRING_CARVING_MASKS = NbtWriter.getStringBytes("CarvingMasks");
+    private static final byte[] STRING_HEIGHTMAPS = NbtWriter.getStringBytes("Heightmaps");
+    private static final byte[] STRING_POST_PROCESSING = NbtWriter.getStringBytes("PostProcessing");
+    private static final byte[] STRING_BLOCK_TICKS = NbtWriter.getStringBytes("block_ticks");
+    private static final byte[] STRING_FLUID_TICKS = NbtWriter.getStringBytes("fluid_ticks");
+    private static final byte[] STRING_STRUCTURES = NbtWriter.getStringBytes("structures");
+    private static final byte[] STRING_STARTS = NbtWriter.getStringBytes("starts");
+    private static final byte[] STRING_BIG_REFERENCES = NbtWriter.getStringBytes("References");
+    private static final byte[] STRING_ID = NbtWriter.getStringBytes("id");
+    private static final byte[] STRING_CHUNK_X = NbtWriter.getStringBytes("ChunkX");
+    private static final byte[] STRING_CHUNK_Z = NbtWriter.getStringBytes("ChunkZ");
+    private static final byte[] STRING_SMALL_REFERENCES = NbtWriter.getStringBytes("references");
+    private static final byte[] STRING_CHILDREN = NbtWriter.getStringBytes("Children");
+    private static final byte[] STRING_INVALID = NbtWriter.getStringBytes("INVALID");
+    private static final byte[] STRING_BB = NbtWriter.getStringBytes("BB");
+    private static final byte[] STRING_O = NbtWriter.getStringBytes("O");
+    private static final byte[] STRING_GD = NbtWriter.getStringBytes("GD");
+    private static final byte[] STRING_NAME = NbtWriter.getStringBytes("Name");
+    private static final byte[] STRING_PROPERTIES = NbtWriter.getStringBytes("Properties");
+
+    private static final byte[] STRING_CHAR_BIG_Y = NbtWriter.getStringBytes("Y");
+    private static final byte[] STRING_CHAR_SMALL_I = NbtWriter.getStringBytes("i");
+    private static final byte[] STRING_CHAR_SMALL_P = NbtWriter.getStringBytes("p");
+    private static final byte[] STRING_CHAR_SMALL_T = NbtWriter.getStringBytes("t");
+    private static final byte[] STRING_CHAR_SMALL_X = NbtWriter.getStringBytes("x");
+    private static final byte[] STRING_CHAR_SMALL_Y = NbtWriter.getStringBytes("y");
+    private static final byte[] STRING_CHAR_SMALL_Z = NbtWriter.getStringBytes("z");
+
+    private static final byte[] STRING_C2ME = NbtWriter.getStringBytes("C2ME");
+    private static final byte[] STRING_KROPPEB = NbtWriter.getStringBytes("Kroppeb was here :); Version: 0.2.2");
```

Comment:
maybe consider spiting the kroppeb comment and version into separate fields for ease of parsing in the future should that ever be necessary